app/grandchallenge/workstations/static/workstations/js/session-control.mjs

```diff
@@ -80,6 +80,8 @@ function sendSessionControlMessage(targetWindow, origin, action, ackCallback) {
         }
         if (ack.id === messageId) {
             ackCallback();
+            targetWindow.focus(); // absolutely necessary in Firefox, in Chrome/Edge window.open() already focusses the window
+            removeSpinner(triggeringElement);
```

Comment:
Can't you place this in the ackCallback() ?

Reply:
The ackCallback just clears the timer, right So then that code would be run if I understand it correctly.

Reply:
It does now, yes. But it can easily be extended. Adding these lines 'dirties' the sendSessionControlMessage function and makes it have multiple responsibilities.